Fix: Assertion failure for SlothMUD on Windows

mpconley opened the pull request:
#### Brief overview of PR changes/additions

• **Fixed assertion failure** by adding comprehensive bounds checking to `processTelnetCommand()` that prevents crashes when processing malformed telnet commands from MCCP2 decompression
• **Enhanced debug output** with hex representation and decoded command information for short/corrupted telnet commands using new `formatShortTelnetCommand()` helper function
• **Defensive programming approach** ensures Mudlet remains stable when receiving truncated network data

#### Motivation for adding to Mudlet

Fixes crash when connecting to SlothMUD on Windows and potentially other MUDs using MCCP2 compression. The issue occurs when MCCP2 decompression produces truncated telnet command sequences, causing out-of-bounds array access. The enhanced debug output makes it much easier to diagnose similar protocol issues in the future by showing both raw hex bytes and decoded command information for malformed commands.

#### Other info (issues closed, discussion etc)

Closes #8104

• **Root cause**: Missing bounds checking in telnet command processing after MCCP2 decompression  
• **Comprehensive fix**: Added bounds checking for all telnet command types (TN_WILL, TN_WONT, TN_DO, TN_DONT, TN_SB)  
• **Improved debugging**: Short commands now display hex + decoded format (e.g., "hex: ff fb (IAC WILL <missing option>)")  
• **No breaking changes**: All existing functionality preserved, purely defensive additions

mpconley committed:
initial

vadi2 commented:
Just a friendly reminder that this PR is in draft mode and it not yet ready for reviewing.

vadi2 commented on src/ctelnet.cpp:
```diff
@@ -1641,6 +1641,12 @@ void cTelnet::autoEnableMXPProcessor()
 {
+    // Ensure telnetCommand has sufficient length before accessing indices
+    if (telnetCommand.size() < 2) {
+        qDebug() << "WARNING: telnetCommand too short (size:" << telnetCommand.size() << "), ignoring";
```
It would be great for the command to be shown as well so you can identify it right away - here and in other places.

mpconley replied:
Done

mpconley committed:
Merge branch 'development' into slothmud_assert

vadi2 commented on test/CTelnetTest.cpp:
```diff
@@ -0,0 +1,237 @@
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include <QtTest/QtTest>
+#include <string>
```
This test just simulates a problem and simulates fixing a problem, without touching Mudlet code. I'm not sure that is useful.

Let's take the fix without it, as indeed the telnet/host classes aren't that well setup for test scenarios.

mpconley replied:
I removed the test.

mpconley committed:
Merge branch 'development' into slothmud_assert

mpconley committed:
Remove CTelnetTest per code review feedback

Per review comment, the test simulates rather than tests actual Mudlet code
and the telnet/host classes aren't well set up for test scenarios.
Keeping only the essential bounds checking fix in ctelnet.cpp.

mpconley committed:
Enhance debug output for short telnet commands

- Add formatShortTelnetCommand helper function for consistent debug formatting
- Show both hex representation and decoded command information
- Improves debugging of malformed telnet commands from MCCP2 or network issues
- Clean, maintainable solution that avoids code duplication
